src/core/storage/disk.test.js

```diff
@@ -0,0 +1,69 @@
+// @flow
+import {DiskStorage} from "./disk";
+import tmp from "tmp";
+import fs from "fs-extra";
+import {join as pathJoin} from "path";
+import {tmpdir} from "os";
+// $FlowIgnore[missing-export]
+import {TextEncoder, TextDecoder} from "util";
+
+describe("core/storage/disk", () => {
+  function tmpWithContents(contents: mixed) {
+    const name = tmp.tmpNameSync();
+    fs.writeFileSync(name, JSON.stringify(contents));
+    return name;
+  }
+  const badPath = () => pathJoin(tmp.dirSync().name, "not-a-real-path");
+  const fooDefault = () => ({foo: 1337});
+  const getInstance = (baseDir?: string) => new DiskStorage(baseDir || "");
+  const basedir = tmp.dirSync();
+  const encoder = new TextEncoder();
+  // $FlowIgnore[incompatible-call]
+  const decoder = new TextDecoder();
+
+  it("DiskStorage accepts a basepath", () => {
+    const instance = new DiskStorage(tmpdir());
+    expect(instance._basePath).toBe(tmpdir());
```

Comment:
Please don’t poke into the private implementation details for tests.
What behavior are you trying to test here? Can it be observed through
the public interface? If so, observe it; if not, by definition, it
doesn’t matter.